ansible/verify_config.py

```diff
@@ -0,0 +1,466 @@
+        self.assertion.assert_true(lambda: self._assertion_fn(var),
+                                   reason=f"Key '{Formatting.red('docker_registry_host')}' must be defined")
+
+    def _assertion_fn(self, var: Dict[str, str]):
+        required_key = 'docker_registry_host'
+        return required_key in var and var[required_key] is not None
+
+
+class TestbedValidator(Validator):
+    def __init__(self):
+        super().__init__(Config.TESTBED_FILE.value)
+
+    def validate(self):
+        conf_name_check_unique = set()
+        group_name_check = {}
+        conf_name_to_vm_range: Dict[str, VMRange] = {}
+
+        for testbed in Utility.parse_yml(self.file):
+            conf_name = testbed['conf-name']
+
+            self.assertion.assert_true(lambda: self._required_attributes_must_be_in_testbed(testbed),
+                                       reason=f"Required attributes must be in testbed '{Formatting.red(conf_name)}'")
+            self.assertion.assert_true(lambda: conf_name not in conf_name_check_unique,
+                                       reason=f"Config name conf_name='{Formatting.red(conf_name)}' must be unique")
+            self.assertion.assert_true(lambda: len(testbed['group-name']) <= 8,
+                                       reason=f"Group name '{Formatting.red(testbed['group-name'])}' must be up to 8 "
+                                              f"characters long. Actual length: {len(testbed['group-name'])}")
+            self.assertion.assert_true(lambda: self._group_name_must_have_unique_attributes(group_name_check, testbed),
+                                       reason=f"Group name '{Formatting.red(testbed['group-name'])}' of "
+                                              f"'{Formatting.red(conf_name)}' does not have unique attributes")
+            self.assertion.assert_true(lambda: testbed['topo'] in Utility.get_topo_from_var_files(),
+                                       reason=f"Topology name '{Formatting.red(testbed['topo'])}' is not "
+                                              f"declared in '{Config.TOPO_FILE_PATTERN.value}'")
+            self.assertion.assert_true(lambda: self._topo_name_must_be_in_veos_file(testbed),
+                                       reason=f"Topology name '{Formatting.red(testbed['topo'])}' is not "
+                                              f"declared in '{Config.VEOS_FILE.value}'")
+            self.assertion.assert_true(lambda: self._server_name_must_be_in_veos_file(testbed),
+                                       reason=f"Server name '{Formatting.red(testbed['server'])}' is not "
+                                              f"declared in '{Config.VEOS_FILE.value}'")
+
+            if testbed['vm_base']:
+                self.assertion.assert_true(lambda: Utility.get_num_vm(testbed['topo']) != 0,
+                                           reason=f"Topology '{Formatting.red(testbed['topo'])}' does not declared to "
```

Comment:
Should be `is not declared to` instead of `does not declared to`?

Reply:
Good pick!